Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

```diff
@@ -207,14 +216,263 @@ class DefaultRiskLevels @Inject constructor(
+        riskCalculationParameters: RiskCalculationParametersOuterClass.RiskCalculationParameters
+    ): AggregatedRiskResult {
+        val uniqueDates = exposureWindowsAndResult.keys
+            .map { it.dateMillisSinceEpoch }
+            .toSet()
+        Timber.d("uniqueDates: ${TextUtils.join(System.lineSeparator(), uniqueDates)}")
+
+        val exposureHistory = uniqueDates.map {
+            exposureDataMapper(
+                it,
+                exposureWindowsAndResult,
+                riskCalculationParameters
+            )
+        }
+
+        exposureHistory.forEach {
+            Timber.d(
+                "(date=%s," +
+                    " riskLevel=%s," +
+                    " minimumDistinctEncountersWithLowRisk=%s," +
+                    " minimumDistinctEncountersWithHighRisk=%s)",
+                it.date,
+                it.riskLevel,
+                it.minimumDistinctEncountersWithLowRisk,
+                it.minimumDistinctEncountersWithHighRisk
+            )
+        }
+
+        // 6. Determine `Total Risk`
+        val totalRiskLevel =
+            if (exposureHistory.any {
+                    it.riskLevel == RiskCalculationParametersOuterClass
+                        .NormalizedTimeToRiskLevelMapping
+                        .RiskLevel
+                        .HIGH
+                }) {
+                RiskCalculationParametersOuterClass.NormalizedTimeToRiskLevelMapping.RiskLevel.HIGH
+            } else {
+                RiskCalculationParametersOuterClass.NormalizedTimeToRiskLevelMapping.RiskLevel.LOW
+            }
+
+        Timber.d("totalRiskLevel: ${totalRiskLevel.name} (${totalRiskLevel.ordinal})")
+
+        // 7. Determine `Date of Most Recent Date with Low Risk`
+
+        // 8. Determine `Date of Most Recent Date with High Risk`
+
+        // 9. Determine `Total Minimum Distinct Encounters With Low Risk`
+
+        // 10. Determine `Total Minimum Distinct Encounters With High Risk`
+
+        TODO("Implement last steps and adjust values")
+
+        return AggregatedRiskResult(
+            totalRiskLevel,
+            totalMinimumDistinctEncountersWithLowRisk = 1,
+            totalMinimumDistinctEncountersWithHighRisk = 1,
+            mostRecentDateWithLowRisk = 1,
+            mostRecentDateWithHighRisk = 1
+        )
+    }
+
+    private fun exposureDataMapper(
+        date: Long,
+        exposureWindowsAndResult: Map<ExposureWindow, RiskResult>,
+        riskCalculationParameters: RiskCalculationParametersOuterClass.RiskCalculationParameters
+    ): ExposureData {
+        // 1. Group `Exposure Windows by Date`
+        val exposureWindowsAndResultForDate = exposureWindowsAndResult
+            .filter { it.key.dateMillisSinceEpoch == date }
+
+        // 2. Determine `Normalized Time per Date`
+        val normalizedTime = exposureWindowsAndResultForDate.values
+            .sumOf { it.normalizedTime }
+
+        Timber.d("normalizedTime: $normalizedTime")
```

Comment:
might be worth logging the date for this aggregated normalized time

Reply:
normalizedTime is no Date or something. what would you recommend to log additonally?

Reply:
This is the normalized time for a certain date, right? when you look into the log, you would just see lots of "normalizedTime: xyz" entries but wouldn't know for which date. I would suggest to just log the `date` param along with this (and for all other log entries in the function/method)

Reply:
Now I got your intension :bulb:
I adjusted it to show a more clear log statement with corresponding Instant now.